store/postgres/user_repository.go

```diff
@@ -32,6 +33,11 @@ func (r *UserRepository) create(ctx context.Context, querier sqlx.QueryerContext
 		return "", err
 	}
 
+	// TODO: provider name is optional for now and we force the default
+	if ud.Provider == "" {
+		ud.Provider = r.cfg.IdentityProviderDefaultName
```

Comment:
if setting a Provider value is business logic, let's keep it in the service instead of in the repository? CMIIW

Reply:
We did assign it in service layer and keep it mandatory in repository layer before.

But yeah perhaps better to revert it back to service layer.